packages/editor/src/components/browser/role/RoleBrowser.tsx

```diff
@@ -92,10 +92,26 @@ const RoleBrowser = (props: {
     props.onChange({ cursorValue: selectedRow.original.id });
   }, [props, rowSelection, table]);
 
+  const [addedRole, setAddedRoleName] = useState('');
+
+  useEffect(() => {
+    const newRow = table.getRowModel().flatRows.find(row => row.original.id === addedRole);
+    if (newRow) {
+      const parentRows = newRow.getParentRows();
+      const expandedState = parentRows.reduce((acc, parentRow) => {
+        acc[parentRow.id] = true;
+        return acc;
+      }, {} as Record<string, boolean>);
+
+      setExpanded(expandedState);
```

Comment:
Techically the parent is already selected, so at least to it everything is expanded right?
So then isn't it enough to make sure this row is expanded?

```suggestion
      newRow.getParentRow().toggleExpanded(true);
```

Reply:
Yes, you're right, thank you. It works the same way and is much easier 👍